src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/mapper/CourtDetailsMapperTest.java

```diff
@@ -237,15 +226,57 @@ void givenCaseDataWithNoPreviousCourtList_whenLondonCourt_thenCaseDataWithOnlyOn
-            is(equalTo(LondonCourt.CENTRAL_FAMILY_COURT)));
-        assertThat(allocatedRegionWrapperReturn.getWalesFrcList(),
-            is(nullValue()));
-        assertThat(allocatedRegionWrapperReturn.getDefaultCourtListWrapper().getSwanseaCourtList(),
-            is(nullValue()));
+        assertEquals(Region.LONDON, allocatedRegionWrapperReturn.getRegionList());
+        assertEquals(RegionLondonFrc.LONDON, allocatedRegionWrapperReturn.getLondonFrcList());
+        assertEquals(LondonCourt.CENTRAL_FAMILY_COURT, allocatedRegionWrapperReturn.getDefaultCourtListWrapper().getLondonCourtList());
+        assertNull(allocatedRegionWrapperReturn.getWalesFrcList());
+        assertNull(allocatedRegionWrapperReturn.getDefaultCourtListWrapper().getSwanseaCourtList());
+    }
+
+    @Test
+    void givenValidCourtField_whenConvertToFrcCourtDetails_thenReturnExpectedCourtDetailsTemplateFields() {
+        // Mocking the court details map
+        when(courtDetailsConfiguration.getCourts()).thenReturn(Map.of(
+            "FR_s_CFCList_2", new CourtDetails("Croydon County Court And Family Court",
+                "Croydon County Court, Altyre Road, Croydon, CR9 5AB", "0300 123 5577", "FRCLondon@justice.gov.uk")
+        ));
+
+        // Setting up the test data
+        DefaultCourtListWrapper courtList = new DefaultCourtListWrapper();
+        courtList.setCfcCourtList(CfcCourt.CROYDON_COUNTY_COURT_AND_FAMILY_COURT);
+
+        Court court = Court
+            .builder()
+            .londonList(RegionLondonFrc.LONDON)
+            .region(Region.LONDON)
+            .courtListWrapper(DefaultCourtListWrapper
+                .builder()
+                .cfcCourtList(CfcCourt.CROYDON_COUNTY_COURT_AND_FAMILY_COURT)
+                .build())
+            .build();
+
+        // Invoking the method
+        CourtDetails courtDetails = courtDetailsMapper.convertToFrcCourtDetails(court);
+
+        // Assertions
+        assertEquals("Croydon County Court And Family Court", courtDetails.getCourtName());
+        assertEquals("Croydon County Court, Altyre Road, Croydon, CR9 5AB", courtDetails.getCourtAddress());
+        assertEquals("FRCLondon@justice.gov.uk", courtDetails.getEmail());
+        assertEquals("0300 123 5577", courtDetails.getPhoneNumber());
+    }
+
+    @Test
+    void givenNoValidFieldInCourtListWrapper_whenConvertToFrcCourtDetails_thenThrowIllegalStateException() {
+        // Mocking a Court object with an empty DefaultCourtListWrapper
+        Court court = Court.builder()
+            .courtListWrapper(DefaultCourtListWrapper.builder().build())
+            .build();
+
+        // Expecting an IllegalStateException
+        IllegalStateException exception = assertThrows(IllegalStateException.class,
+            () -> courtDetailsMapper.convertToFrcCourtDetails(court));
+
+        // Verifying the exception message
+        assertEquals("No valid field found in the court list wrapper", exception.getMessage());
```

Comment:
Reading the test, it seems like it would be handy to see something about the 'court' being tried here.
Would really help support if this happened in Prod.

Reply:
your court instance, passed into the function, is the only thing you could log:

("A message that it went wrong: {}", court);

Might be useful.  However, I take your point, it's tricky.